Cli/Command/CheckViolationsCommand.php

```diff
@@ -0,0 +1,155 @@
+            ->addOption('no-global'     , null, InputOption::VALUE_NONE, 'Disable showing global violations', null)
+            ->addOption('ignore-assets' , null, InputOption::VALUE_NONE, 'Ignore assets violations(*.css, *.js, *.coffee, *.less, *.sass)', null)
+            ->setDescription('Check for violations occurred in one or multiple commits in the last analysis.')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $projectUuid = $input->getArgument('project-uuid');
+        $api = $this->getApplication()->getApi();
+        $gitRepository = new GitRepository($input->getOption('repository'));
+
+        if ($gitRepository->isBare()) {
+            throw new \Exception(
+                sprintf('No Git repository found on this path "%s"', $input->getOption('repository'))
+            );
+        }
+
+        $output->writeln(sprintf('<info>Found GIT repository in <%s></info>', $gitRepository->getGitDir()));
+
+        if ($input->getOption('commits')) {
+            $revisionsList = new GitRevisionList($gitRepository, $input->getOption('commits'));
+            $diff = $gitRepository->getDiff($revisionsList);
+
+        } else {
+            $diff = $gitRepository->getHeadCommit()->getDiff();
+        }
+
+        if (!$diff) {
+            $output->writeln('<info>No diff found.</info>');
+
+            return;
+        }
+
+        $changedFiles = array();
+        foreach ($diff->getFiles() as $file) {
+            if ($file->getNewName()) {
+                if (!$input->getOption('ignore-assets') && $this->isAsset($file->getNewName())) {
+                    continue;
+                }
+
+                $changedFiles[] = $file->getNewName();
+            }
+        }
+
+        $output->writeln(sprintf('<info>Found [%d] changed files</info>', count($changedFiles)));
+        $output->writeln(sprintf('<info>Get violations from latest SensioLabs insight analysis...</info>', count($changedFiles)));
+
+        $sensioInsightAnalysis = $api->getProject($input->getArgument('project-uuid'))->getLastAnalysis();
+        $analysisLinkPage = sprintf('%s/projects/%s/analyses/%d', $api::ENDPOINT, $projectUuid, $sensioInsightAnalysis->getNumber());
+
+        $output->writeln(
+            sprintf('<info>Last analysis number is [%d], link page <%s></info>', $sensioInsightAnalysis->getNumber(), $analysisLinkPage)
+        );
+
+        if (!($sensioInsightAnalysis->getEndAt() instanceof \DateTime)) {
+            $output->writeln(
+                sprintf(
+                    '<bg=yellow;fg=red>The analysis number [%d], started at [%s] is not yet finished</bg=yellow;fg=red>',
+                    $sensioInsightAnalysis->getNumber(),
+                    $sensioInsightAnalysis->getBeginAt()->format('d D Y, H:i:s')
+                )
+            );
+
+            return;
```

Comment:
This is how the output is returned, I think it's good rather than exception ? ![selection_024](https://cloud.githubusercontent.com/assets/761378/7937038/55009f54-0935-11e5-94cd-fdf1df5b7986.png)